### PR TITLE
Provide a way to enter a line total and have the quantity sold automatically calculated.

### PR DESCRIPTION
**Pull request by SteveIreland**
This should result in the behavior last discussed.

### CODE REVIEW

File: application/views/sales/register.php [hunk 1]
```diff
@@ -179,11 +179,11 @@
 								<?php
 								if($item['item_type'] == 2)
 								{
-									echo form_input(array('name'=>'total', 'class'=>'form-control input-sm', 'value'=>to_currency_no_money($item['total']), 'tabindex'=>++$tabindex));
+									echo form_input(array('name'=>'discounted_total', 'class'=>'form-control input-sm', 'value'=>to_currency_no_money($item['discounted_total']), 'tabindex'=>++$tabindex));
 								}
 								else
 								{
-									echo to_currency($item['price']*$item['quantity']-$item['price']*$item['quantity']*$item['discount']/100);
```
Comment (daN4cat) [inline application/views/sales/register.php]: I'm afraid I cannot get where this is now calculated. Your changes don't show this calculation being moved somewhere else. Or am I missing something?

Comment (SteveIreland) [reply application/views/sales/register.php]: Sale_lib.php line 903

$line['discounted_total'] = $this->get_item_total($quantity, $price, $discount, TRUE);

Comment (daN4cat) [reply application/views/sales/register.php]: Ok, got the point

**Comment by SteveIreland** [comment]
Okay, I've removed the statements that were attempting to check for a required field on the extended total.  To have properly corrected it I would have needed to add an item info retrieval to the method.  I didn't think it was worth the overhead so I just removed the code.  If the user inexplicably enters a zero amount then the quantity will be set to zero. I think it will be okay.

**Pull request merged**